OpenRA.Mods.Common/Traits/SupportPowers/SelectDirectionalTarget.cs

```diff
@@ -117,37 +107,26 @@ bool IsOutsideDragZone
 
 		IEnumerable<IRenderable> IOrderGenerator.RenderAboveShroud(WorldRenderer wr, World world)
 		{
-			if (!activated)
-				return Enumerable.Empty<IRenderable>();
-
-			var targetPalette = wr.Palette(targetPlaceholderCursorPalette);
+			if (!activated || !IsOutsideDragZone)
+				yield break;
 
-			var location = activated ? targetLocation : Viewport.LastMousePos;
-			var worldPx = wr.Viewport.ViewToWorldPx(location);
+			var worldPx = wr.Viewport.ViewToWorldPx(targetLocation);
 			var worldPos = wr.ProjectedPosition(worldPx);
-			var renderables = new List<IRenderable>(targetCursor.Render(worldPos, WVec.Zero, -511, targetPalette, 1 / wr.Viewport.Zoom));
-
-			if (IsOutsideDragZone)
-			{
-				var directionPalette = wr.Palette(directionArrowPalette);
-				renderables.Add(new SpriteRenderable(currentArrow.Sprite, worldPos, WVec.Zero, -511, directionPalette, 1 / wr.Viewport.Zoom, true));
-			}
 
-			if (hideMouse)
-			{
-				hideMouse = false;
-				Game.RunAfterTick(() => Game.HideCursor = true);
-			}
+			var scale = 1 / wr.Viewport.Zoom;
+			if (!Game.Settings.Graphics.HardwareCursors)
```

Comment:
Another edge case from testing: start a game with hardware cursors, then switch to software cursors in the settings. The cursor will stay normally sized (the actual switch requires a restart) but the direction arrows will be doubled.

How about
```csharp
// Note: We can't trust the GraphicsSettings.HardwareCursors because changes are ignored until the next game launch
var scale = (Game.Cursor is SoftwareCursor && CursorProvider.CursorViewportZoomed ? 2 : 1) / wr.Viewport.Zoom;
```
?